### ISSUE TITLE
Any technique for hyperparameter tuning? 

### ISSUE THREAD

**Issue by NitinAB1108** [princeton-nlp/SimPO#50]
I can't seem to effectively train models other than the ones mentioned in the configs. Any technique to arrive at effective gamma, beta and learning rate for a given LLM and a dataset? Or is it just trial and error?

**Comment by yumeng5** [comment]
Hi @NitinAB1108 

We provided tips on hyperparameter tuning here: https://github.com/princeton-nlp/SimPO?tab=readme-ov-file#hyperparameter-tuning

We recommend tuning `learning_rate`, `beta`, and `gamma` one by one instead of simultaneously: Start by tuning the `learning_rate` while keeping `beta` and `gamma` at their default values (e.g., `beta = 10` and `gamma_beta_ratio = 0.25`). Once you have identified a good `learning_rate`, proceed to tune `gamma_beta_ratio`, keeping `learning_rate` and `beta` constant. Finally, you can optionally fine-tune `beta`, although we have found that `beta = 10` generally performs well across various datasets and LLMs.

I hope this helps.

Best,
Yu

**Comment by NitinAB1108** [comment]
Thanks, that helped.

Best,
Nitin
